flaskbb/forum/views.py

```diff
@@ -627,7 +707,18 @@ def post(self):
 
 
 class DeleteTopic(MethodView):
-    decorators = [allows.requires(CanDeleteTopic), login_required]
+    decorators = [
+        allows.requires(
+            CanDeleteTopic,
+            on_fail=FlashAndRedirect(
+                message=_("You are not allowed to delete this topic"),
+                level="danger",
+                # TODO(anr): consider the referrer -- for now, back to topic
```

Comment:
There is a case -- should only be possible in corner cases and purposefully bad actors -- where the user is redirected back to the topic and then back to the category because they're not allowed to access the forum and get a couple of warning messages saying they're not allowed to delete the topic, enter the thread and enter the forum. This happens for all the moderation actions in that case.

As long as we're aware of it it's probably not a big deal.

Reply:
oh duh, can access forum. that'll only do one redirect